File: classify-server/src/main/java/io/ermdev/classify/data/repository/UserRepository.java
package io.ermdev.classify.data.repository;

import io.ermdev.classify.data.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface UserRepository extends JpaRepository<User, Long> {

    @Query("from User where id=:userId")
    User findById(@Param("userId") Long userId);

    @Query("from User")
    List<User> findAll();
}

File: classify-spring-boot-webservices/src/main/kotlin/io/ermdev/classify/data/repository/UserRepository.kt
package io.ermdev.classify.data.repository

import io.ermdev.classify.data.entity.Student
import io.ermdev.classify.data.entity.Teacher
import io.ermdev.classify.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import java.util.*

interface UserRepository : JpaRepository<User, Long> {

    @Query("from User where username=:username")
    fun findByUsername(@Param("username") username: String): Optional<User>

    @Query("from Student as a join a.user as b where b.id=:userId")
    fun findStudent(@Param("userId") userId: Long): Student?

    @Query("from Teacher as a join a.user as b where b.id=:userId")
    fun findTeacher(@Param("userId") userId: Long): Teacher?
}